File: frontend/types/Responses/Document.ts
export interface DocumentResponse {
  id: number;
  document_name: string;
  owner_email: string;
  created_at: string;
  stage?: number | null;
  completion_stage?: number | null;
}

export interface DocumentEntity {
  id: number;
  document_name: string;
  owner_email?: string;
  stage?: number | null;
  completion_stage?: number | null;
  created_at?: string;
}

export interface UpdateCompletionStageResponse {
  id: number;
  completion_stage: number;
  updated_at: string;
}

File: backend/app/schemas/document.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class DocumentBase(BaseModel):
    document_name: str
    stage: int

class DocumentCreate(BaseModel):
    document_name: str
    user_id: Optional[int] = None
    stage: int

class DocumentUpdate(BaseModel):
    document_name: Optional[str] = None
    stage: Optional[int] = None

class DocumentRead(DocumentBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    deleted_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class CompletionStageUpdate(BaseModel):
    completion_stage: int
